bpf/go_common.h

```diff
@@ -24,13 +36,50 @@ struct {
     __uint(max_entries, 1 << 24);
 } events SEC(".maps");
 
+
+// Temporary information about a function invocation. It stores the invocation time of a function
+// as well as the value of registers at the invocation time. This way we can retrieve them at the
+// return uprobes so we can know the values of the function arguments (which are passed as registers
+// since Go 1.17).
+// This element is created in the function start probe and stored in the ongoing_http_requests hashmaps.
+// Then it is retrieved in the return uprobes and used to know the HTTP call duration as well as its
+// attributes (method, path, and status code).
+typedef struct func_invocation_t {
+    u64 start_monotime_ns;
+    struct pt_regs regs; // we store registers on invocation to be able to fetch the arguments at return
+} func_invocation;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __type(key, void *); // key: pointer to the request goroutine
+    __type(value, func_invocation);
+    __uint(max_entries, MAX_CONCURRENT_REQUESTS);
+} newproc1 SEC(".maps");
+
+typedef struct goroutine_metadata_t {
+    u64 parent;
+    u64 timestamp;
+} goroutine_metadata;
+
 struct {
     __uint(type, BPF_MAP_TYPE_HASH);
     __type(key, void *); // key: pointer to the goroutine
-    __type(value, u64);  // value: timestamp of the goroutine creation
+    __type(value, goroutine_metadata);  // value: timestamp of the goroutine creation
     __uint(max_entries, MAX_CONCURRENT_REQUESTS);
+	__uint(pinning, LIBBPF_PIN_BY_NAME);
```

Comment:
nit: adjust indentation?

Reply:
Maybe after this PR is merged, we should also evaluate whether we will have any type of conflict/corruption if we use the same BPF map from multiple auto-instrumenters in the same host.

If that happened, maybe we can try to assign a unique per-process map name on load so they are not shared (it seems Cilium library would allow that).

Another option is to override the `/sys/fs/bpf` path a unique `/var/run/otelauto/...` subfolder. [LibBPF allows doing it](https://elixir.bootlin.com/linux/latest/source/tools/lib/bpf/libbpf.c#L2118) but I don't know if Cilium lib can.